packages/field-base/test/delegate-focus-mixin.test.js

```diff
@@ -0,0 +1,139 @@
+      element.blur();
+      expect(spy.calledOnce).to.be.true;
+    });
+
+    it('should remove focused attribute on blur call', () => {
+      element.focus();
+      element.blur();
+      expect(element.hasAttribute('focused')).to.be.false;
+    });
+
+    it('should not focus the input on focus when disabled', () => {
+      const spy = sinon.spy(input, 'focus');
+      element.disabled = true;
+      element.focus();
+      expect(spy.calledOnce).to.be.false;
+    });
+
+    it('should not set focused attribute when disabled', () => {
+      element.disabled = true;
+      element.focus();
+      expect(element.hasAttribute('focused')).to.be.false;
+    });
+
+    it('should propagate disabled property to the input', () => {
+      element.disabled = true;
+      expect(input.disabled).to.be.true;
+
+      element.disabled = false;
+      expect(input.disabled).to.be.false;
+    });
+
+    it('should call blur when disabled is set to true', () => {
+      const spy = sinon.spy(element, 'blur');
+      element.disabled = true;
+      expect(spy.calledOnce).to.be.true;
+    });
+
+    it('should click the input on click call', () => {
+      const spy = sinon.spy(input, 'click');
+      element.click();
+      expect(spy.calledOnce).to.be.true;
+    });
+
+    it('should not click the input on click when disabled', () => {
+      const spy = sinon.spy(input, 'click');
+      element.disabled = true;
+      element.click();
+      expect(spy.calledOnce).to.be.false;
+    });
+  });
+
+  describe('autofocus', () => {
+    beforeEach(() => {
+      element = document.createElement('delegate-focus-mixin-element');
+      element.autofocus = true;
+    });
+
+    afterEach(() => {
+      document.body.removeChild(element);
+    });
+
+    it('should focus the input when autofocus is set', async () => {
+      document.body.appendChild(element);
+      const spy = sinon.spy(element._inputNode, 'focus');
+      await nextFrame();
+      expect(spy.calledOnce).to.be.true;
+    });
+
+    it('should set focused attribute when autofocus is set', async () => {
+      document.body.appendChild(element);
+      await nextFrame();
+      expect(element.hasAttribute('focused')).to.be.true;
+    });
+
+    it('should set focus-ring attribute when autofocus is set', async () => {
+      document.body.appendChild(element);
+      await nextFrame();
+      expect(element.hasAttribute('focus-ring')).to.be.true;
+    });
+
+    it('should not focus the input when disabled is set', async () => {
+      element.disabled = true;
+      document.body.appendChild(element);
+      const spy = sinon.spy(element.focusElement, 'focus');
+      await nextFrame();
+      expect(spy.called).to.be.false;
+    });
+
+    it('should not set focused attribute when autofocus is set', async () => {
+      element.disabled = true;
+      document.body.appendChild(element);
+      await nextFrame();
+      expect(element.hasAttribute('focused')).to.be.false;
+    });
```

Comment:
Adding another test asserting that `focus-ring` doesn't set on the disabled element might be appropriate.

Reply:
Done.